### PR TITLE
Feature/target pose estimation.

### PR DESCRIPTION
**Pull request by fabianbl**
Target pose estimation for an AprilTag grid (and potentially any target grid).

### CODE REVIEW

File: aslam_cv_calibration/src/target-algorithms.cc [hunk 1]
```diff
@@ -0,0 +1,51 @@
+#include "aslam/calibration/target-algorithms.h"
+
+#include <Eigen/Core>
+#include <aslam/geometric-vision/pnp-pose-estimator.h>
+
+namespace aslam {
+namespace calibration {
+
+bool estimateTargetTransformation(
```
Comment (floriantschopp) [inline aslam_cv_calibration/src/target-algorithms.cc]: Why not using a single function with [default arguments](http://en.cppreference.com/w/cpp/language/default_arguments)?

File: aslam_cv_calibration/src/target-algorithms.cc [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+    const aslam::Camera::ConstPtr& camera_ptr,
+    const bool run_nonlinear_refinement, const double ransac_pixel_sigma,
+    const int ransac_max_iters, aslam::Transformation* T_G_C) {
+  CHECK(camera_ptr);
+  CHECK_GT(ransac_pixel_sigma, 0.0);
+  CHECK_GT(ransac_max_iters, 0);
+  CHECK_NOTNULL(T_G_C);
+  const Eigen::Matrix2Xd& observed_corners =
+      target_observation.getObservedCorners();
+  const Eigen::Matrix3Xd G_corner_positions =
```
Comment (floriantschopp) [inline aslam_cv_calibration/src/target-algorithms.cc]: Maybe rename to corresponding_corner_positions?

File: aslam_cv_calibration/src/target-algorithms.cc [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+  aslam::geometric_vision::PnpPoseEstimator pnp(run_nonlinear_refinement);
+  std::vector<int> inliers;
+  int num_iters = 0;
+  bool pnp_success = pnp.absolutePoseRansacPinholeCam(
+      observed_corners, G_corner_positions, ransac_pixel_sigma,
+      ransac_max_iters, camera_ptr, T_G_C, &inliers, &num_iters);
+  if (pnp_success) {
+    VLOG(4) << "Found " << inliers.size() << "/" << observed_corners.cols()
+            << " inliers in" << num_iters << " iterations.";
+  } else{
```
Comment (floriantschopp) [inline aslam_cv_calibration/src/target-algorithms.cc]: } else {

File: aslam_cv_calibration/include/aslam/calibration/target-base.h [hunk 4]
```diff
@@ -45,11 +45,16 @@ class TargetBase {
...
-  size_t rows() const { return rows_; };
+  inline size_t rows() const { return rows_; };
   /// Number of columns in the calibration target.
-  size_t cols() const { return cols_; };
+  inline size_t cols() const { return cols_; };
   /// Get the number of points of the full grid.
-  size_t size() const { return rows_ * cols_; };
+  inline size_t size() const { return rows_ * cols_; };
+
+  /// Extent of the grid in row dimension (x by default).
```
Comment (floriantschopp) [inline aslam_cv_calibration/include/aslam/calibration/target-base.h]: is x a placeholder?

Comment (fabianbl) [reply aslam_cv_calibration/include/aslam/calibration/target-base.h]: No it should actually denote the x coordinate :-)

File: aslam_cv_calibration/src/target-algorithms.cc (hunk 1 above)
Comment (fabianbl) [reply aslam_cv_calibration/src/target-algorithms.cc]: I think the preferred ASL style is having multiple functions instead of default args.

**Comment by fabianbl** [comment]
@floriantschopp Thanks for the review!

File: aslam_cv_calibration/include/aslam/calibration/target-observation.h [hunk 5]
```diff
@@ -16,32 +16,39 @@
...
+        image_width(im_width),
+        corner_ids_(corner_ids),
+        image_corners_(image_corners) {
     CHECK(target);
     CHECK_EQ(corner_ids.rows(), image_corners.cols());
     buildIndex();
   }
 
-  virtual ~TargetObservation() {};
+  virtual ~TargetObservation(){};
```
Comment (floriantschopp) [inline aslam_cv_calibration/include/aslam/calibration/target-observation.h]: linter

File: aslam_cv_calibration/src/target-algorithms.cc (hunk 1 above)
Comment (floriantschopp) [reply aslam_cv_calibration/src/target-algorithms.cc]: OK lets stick with this then ;-)

**Pull request merged**